ctru-rs/src/services/am.rs

```diff
@@ -56,6 +68,55 @@ impl<'a> Title<'a> {
             Ok(info.assume_init())
         }
     }
+
+    pub fn get_smdh(&self) -> crate::Result<Smdh> {
+        // i have no idea how to make this look better
+        let archive_path_data: [u32; 4] =
+            [self.low_u32(), self.high_u32(), self.mediatype as u32, 0x0];
+        let smdh_path_data: [u32; 5] = [0x0, 0x0, 0x2, u32::from_le_bytes(*b"icon"), 0x0];
+
+        let archive_path = ctru_sys::FS_Path {
+            type_: ctru_sys::PATH_BINARY,
+            size: size_of::<[u32; 4]>() as u32,
+            data: archive_path_data.as_ptr() as *const libc::c_void,
+        };
+        let smdh_path = ctru_sys::FS_Path {
+            type_: ctru_sys::PATH_BINARY,
+            size: size_of::<[u32; 5]>() as u32,
+            data: smdh_path_data.as_ptr() as *const libc::c_void,
+        };
+
+        let _fs = Fs::init();
+        let mut smdh_handle = 0;
+
+        let smdh: Smdh = unsafe {
+            let mut ret = MaybeUninit::zeroed();
+            ResultCode(ctru_sys::FSUSER_OpenFileDirectly(
+                &mut smdh_handle,
+                ctru_sys::ARCHIVE_SAVEDATA_AND_CONTENT,
+                archive_path,
+                smdh_path,
+                ctru_sys::FS_OPEN_READ,
+                0x0,
+            ))?;
+
+            ResultCode(ctru_sys::FSFILE_Read(
+                smdh_handle,
+                std::ptr::null_mut(),
+                0x0,
+                ret.as_mut_ptr() as *mut libc::c_void,
+                size_of::<Smdh>() as u32,
+            ))?;
+
+            ResultCode(ctru_sys::FSFILE_Close(smdh_handle))?;
+
+            ret.assume_init()
+        };
+
+        assert_eq!(&smdh.magic(), b"SMDH");
```

Comment:
Instead of assert could this return `Err` if the magic is wrong ("corrupted SMDH" or something)